### DISCUSSION TITLE
How to make sense of the stmt output?

### DISCUSSION THREAD

**Discussion by Ethanwolfie** [halide/Halide#8428] [Q&A] [answered]
I'm exerimenting using Halide to accelerate some algorithms on arm cpus using autoschedulers. Halide helped me realized some algorithms with very impressive results, but I still need some a little bit further optimization to apply the algorithm to my application. I read that you can gain more understanding of the schedule that the autoschedulers generated by reading the stmt output, but I can't seem to understand the symbols and grammar of the generated pheudo code. How do I learn more about this stuff? Where do I start? 
![image](https://github.com/user-attachments/assets/e07c5370-2c09-469a-a10d-467d2c827c3a)

**Comment by abadams** [comment, marked as answer]
The closures and closure prototypes are an artifact of how we lower parallel work. I suggest using "conceptual_stmt" instead of "stmt". It's at a higher level of abstraction, and parallel for loops just look like loops with the word "parallel" in there somewhere.
